### PR TITLE
Add UPDATE and SELET emitters

### PR DESCRIPTION
**Pull request by solnic**
This adds basic support for update/select statements. It also replaces `Where` node with `Delimited` + `K_WHERE` node. Group by is done in the same way as `where`.

refs #1

### CODE REVIEW

File: lib/sql/generator/emitter/assignment.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# encoding: utf-8
+
+module SQL
+  module Generator
+    class Emitter
+
+      # Assignment emitter
+      class Assignment < self
```
Comment (solnic) [inline lib/sql/generator/emitter/assignment.rb]: @mbj @dkubb are you OK with this name?

File: lib/sql/generator/emitter/columns.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+
+module SQL
+  module Generator
+    class Emitter
+
+      # Column names emitter
+      class Columns < self
+        handle :columns
```
Comment (solnic) [inline lib/sql/generator/emitter/columns.rb]: @dkubb @mbj same question - are you ok with this name?

File: lib/sql/generator/emitter/assignment.rb (hunk 1 above)
Comment (mbj) [inline lib/sql/generator/emitter/assignment.rb]: For now, Yeah.

But `dispatch` is so generic, we'll find another name later once we reuse that emitter!

File: lib/sql/generator/emitter/columns.rb (hunk 2 above)
Comment (mbj) [inline lib/sql/generator/emitter/columns.rb]: @solnic I'm okay with the type name, but not with the duplication in `dispatch`. IMHO `:columns`, and `:assignments` should be handled via a single `Delimited` emitter. `handle :columns, :assignments`.

File: lib/sql/generator/emitter/assignment.rb (hunk 1 above)
Comment (mbj) [inline lib/sql/generator/emitter/assignment.rb]: Lets call the type `:assignments`.

**Comment by solnic** [comment]
@dkubb @mbj lemme know if I can merge it in

**Comment by mbj** [comment]
@solnic Go!

BTW we use `write(';')` multiple times. Lets add a helper and a const later.

**Pull request merged**